src/Sylius/Bundle/CoreBundle/Provider/FastestDeliveryTimeProvider.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Provider;
+
+use Sylius\Component\Channel\Model\ChannelInterface as BaseChannelInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Repository\ShippingMethodRepositoryInterface;
+use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+
+final class FastestDeliveryTimeProvider implements DeliveryTimeProviderInterface
+{
+    public function __construct(
+        private readonly ShippingMethodRepositoryInterface $shippingMethodRepository,
+    ) {
+    }
+
+    /**
+     * @return array{minimumDays:int, maximumDays:int}|null
+     */
+    public function provide(BaseChannelInterface $channel, array $context = []): ?array
+    {
+        if (!$channel instanceof ChannelInterface) {
+            return null;
+        }
+
+        /** @var iterable<ShippingMethodInterface> $methods */
+        $methods = $this->shippingMethodRepository->findEnabledForChannel($channel);
+
+        $best = null;
+
+        foreach ($methods as $method) {
+            $range = $this->normalizeDeliveryRange($method->getMinDeliveryTimeDays(), $method->getMaxDeliveryTimeDays());
+            if ($range === null) {
+                continue;
+            }
+
+            [$minimumDays, $maximumDays] = $range;
+
+            if (
+                $best === null ||
+                $maximumDays < $best['maximumDays'] ||
+                ($maximumDays === $best['maximumDays'] && $minimumDays < $best['minimumDays'])
+            ) {
+                $best = ['minimumDays' => $minimumDays, 'maximumDays' => $maximumDays];
+            }
+        }
+
+        return $best;
```

Comment:
Shipping methods have a `position` field. I think it would be predictable to, in case of the exact same dates, propose the one with the lower position first.

Reply:
I think it’s a good idea for an extension, although I’m not sure if it’s worth adding at this stage since it’s still very basic. What do you think?